src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -389,41 +315,210 @@ private static <T extends Enum<T>> int mappedEnumLookup(T value, MethodHandles.L
+                  mb.withFlags(AccessFlag.PUBLIC, AccessFlag.FINAL, AccessFlag.STATIC)
+                    .withCode(cb -> {
+                        cb.aload(0);
+                        Label nonNullLabel = cb.newLabel();
+                        cb.if_nonnull(nonNullLabel);
+                        cb.iconst_m1();
+                        cb.ireturn();
+                        cb.labelBinding(nonNullLabel);
+                        if (labels.length == 0) {
+                            cb.constantInstruction(0)
+                              .ireturn();
+                            return ;
+                        }
+                        cb.iload(1);
+                        Label dflt = cb.newLabel();
+                        record Element(Label target, Label next, Object label) {}
+                        List<Element> cases = new ArrayList<>();
+                        List<SwitchCase> switchCases = new ArrayList<>();
+                        Object lastLabel = null;
+                        for (int idx = labels.length - 1; idx >= 0; idx--) {
+                            Object currentLabel = labels[idx];
+                            Label target = cb.newLabel();
+                            Label next;
+                            if (lastLabel == null) {
+                                next = dflt;
+                            } else if (lastLabel.equals(currentLabel)) {
+                                next = cases.getLast().next();
+                            } else {
+                                next = cases.getLast().target();
+                            }
+                            lastLabel = currentLabel;
+                            cases.add(new Element(target, next, currentLabel));
+                            switchCases.add(SwitchCase.of(idx, target));
+                        }
+                        cases = cases.reversed();
+                        switchCases = switchCases.reversed();
+                        cb.tableswitch(0, labels.length - 1, dflt, switchCases);
+                        for (int idx = 0; idx < cases.size(); idx++) {
+                            Element element = cases.get(idx);
+                            Label next = element.next();
+                            cb.labelBinding(element.target());
+                            if (element.label() instanceof Class<?> classLabel) {
+                                cb.aload(0);
+                                cb.instanceof_(classLabel.describeConstable().get());
+                                cb.ifeq(next);
+                            } else if (element.label() instanceof EnumDesc<?> enumLabel) {
+                                int enumIdx = enumDescs.size();
+                                enumDescs.add(enumLabel);
+                                cb.aload(2);
+                                cb.constantInstruction(enumIdx);
+                                cb.invokestatic(Integer.class.describeConstable().get(),
```

Comment:
You can use `ConstantDescs.CD_Integer` etc.

Reply:
Done.